### DISCUSSION TITLE
Menu's dropdown on Chart is being offset

### DISCUSSION THREAD

**Discussion by somguyth** [chakra-ui/chakra-ui#4455] [Q&A] [answered]
I'm adding a `chakra-ui` menu dropdown button to a chart (from `react-financial-charts`) which is a library built upon `svg` internally. 

For some reason, when I click on the menu, there will be whitespace between the button and the dropdown menu. This only happens when I put the menu onto the chart. If I have the menu standalone in the browser, it'll work as expected.

[![enter image description here][1]][1]

This is the menu code:

```
function TestMenu() {
  return (
    <g className="react-financial-charts-enable-interaction">
      <foreignObject
        // x={0}
        // y={0}
        width={"100%"}
        height={"100%"}
        style={{ overflow: "auto" }}
      >
        <Menu>
          <MenuButton
            as={Button}
            rightIcon={<ChevronDownIcon />}
            transition="all 0.001s"
            borderRadius="md"
            borderWidth="0px"
            _hover={{ bg: "gray.400" }}
            _expanded={{ bg: "blue.400" }}
            _focus={{ boxShadow: "none" }}
            style={{ marginTop: "30px", marginLeft: "30px" }} // better move the style to css file
          >
            Actions
          </MenuButton>
          <MenuList>
            <MenuItem>Download</MenuItem>
            <MenuItem>Create a Copy</MenuItem>
            <MenuItem>Mark as Draft</MenuItem>
            <MenuItem>Delete</MenuItem>
            <MenuItem>Attend a Workshop</MenuItem>
          </MenuList>
        </Menu>
      </foreignObject>
    </g>
  );
}
```

This is the full codesandbox:

https://codesandbox.io/s/nostalgic-pare-c5rxu?file=/src/BasicLineSeries.tsx

  [1]: https://i.stack.imgur.com/uJRQJ.png

**Comment by payapula** [comment]
I don't understand the use case for rendering Menu inside a vector graphics tag. I am not sure if that is accessible or proper semantics. You can style your menu button with svg icon though.

That being said, I don't see any chopping off on the sandbox repo you have shared.

**Reply by somguyth** [reply]
I've changed the question. I was able to render the full menu (my previous question) by increasing the height of the `svg` box. But I've run into this new snag that I describe in the OP.

**Comment by somguyth** [comment, marked as answer]
I had to wrap `MenuList` inside a `Portal` and set `zIndex={10}` inside the `MenuList` props.
